### PR TITLE
WX-1595 Remove genomics usage from GCP Batch

### PR DESCRIPTION
**Pull request by AlexITC**
The Batch backend configuration depends on entries prefixed by "genomics" which is the wrong term, this entry has been renamed to "batch-api".

Also, the usages for these configs were renamed to "batch".

At last, there was an usage involving lifesciences which has been removed too.

NOTE: This is a breaking change, existing projecs using batch backend will need to update their configuration.

### CODE REVIEW

File: supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/actors/GcpBatchInitializationActor.scala [hunk 1]
```diff
@@ -72,8 +72,8 @@ class GcpBatchInitializationActor(batchParams: GcpBatchInitializationActorParams
   private lazy val gcsCredentials: Future[Credentials] = gcpBatchConfiguration.batchAttributes.auths.gcs
     .retryCredentials(workflowOptions, List(StorageScopes.DEVSTORAGE_FULL_CONTROL))
 
-  // Credentials object for the Genomics API
-  private lazy val genomicsCredentials: Future[Credentials] = gcpBatchConfiguration.batchAttributes.auths.genomics
+  // Credentials object for the Batch API
+  private lazy val batchCredentials: Future[Credentials] = gcpBatchConfiguration.batchAttributes.auths.batch
     .retryCredentials(workflowOptions,
                       List(
                         CloudLifeSciencesScopes.CLOUD_PLATFORM,
```
Comment (AlexITC) [inline supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/actors/GcpBatchInitializationActor.scala:79]: I have tried removing this (`CloudLifeSciencesScopes.CLOUD_PLATFORM `) and `GenomicsScopes.GENOMICS` but centaur tests failed.

Does anyone know whether we should use a different scope? I'd expect that we shouldn't need anything lifesciences/genomics related (see https://github.com/broadinstitute/cromwell/actions/runs/8815748415/job/24198334692).

File: supportedBackends/google/batch/src/test/scala/cromwell/backend/google/batch/actors/GcpBatchInitializationActorSpec.scala [hunk 2]
```diff
@@ -139,12 +139,10 @@ object GcpBatchInitializationActorSpec {
       |  // This is the maximum polling interval (in seconds):
       |  maximum-polling-interval = 600
       |
-      |  genomics {
-      |  // A reference to an auth defined in the `google` stanza at the top.  This auth is used to create
-      |  // Pipelines and manipulate auth JSONs.
+      |  batch {
```
Comment (aednichols) [inline supportedBackends/google/batch/src/test/scala/cromwell/backend/google/batch/actors/GcpBatchInitializationActorSpec.scala:142]: I'd personally prefer we use `batch` everywhere for consistency.

Seems easier to get right, especially when transitioning from a product we just called `genomics`.

**Review by aednichols — CHANGES_REQUESTED** [review]
This could use a `CHANGELOG.md` entry to inform users about the config change. Maybe also explain that Batch support is in Alpha, hence there might be breaking changes.

**Comment by AlexITC** [comment]
@aednichols the comments are resolved and tests are passing, thanks.

**Review by aednichols — APPROVED** [review]

**Review by rsaperst — APPROVED** [review]

**Pull request merged**